### PR TITLE
Fix fusion of `<$` operator

### PR DESCRIPTION
**Pull request by lehins**
I've tried both of the suggested approaches from #248 
```haskell
  {-# INLINE (<$) #-}
  (<$) a v = replicate (length v) a
```

and this one https://github.com/haskell/vector/pull/248#issuecomment-583745720 together with
```haskell
  {-# INLINE (<$) #-}
  (<$) x = G.unstream . (x <$) . G.stream
```

None of them prevent this example from diverging:
```haskell
  let v = V.unfoldr (\x -> Just (x, x + 1)) (0 :: Int)
  print $ V.take 4 ((1 :: Int) <$ v) -- diverges
```

Default implementation for `<$` is identical to the one in this PR, but because it does not have an `INLINE` pragma it also fails to fuse.

Approach in this PR is the only one I found that produces the desired output:
```
[1,1,1,1]
```

In other words, this PR does not change the semantics, but fixes fusion for `<$` operator.

### CODE REVIEW

**Comment by lehins** [comment]
@Shimuuar Done. Added the same implementation for both `Bundle` and `Stream`

**Comment by Shimuuar** [comment]
It looks ready for merge to me. I would wait a bit to let @Bodigrim take a look at it
